File: python/arctan_pi.py
from time import time
import math

N = 1_000_000_000

def calculate_pi() -> float:

    pi = 6 * math.atan( 1 /math.sqrt(3))
    return pi


if __name__ == "__main__":
    start = time()
    pi = calculate_pi()
    end = time()
    print(f"π = {pi:1.38f}")
    # print(f"Time elapsed: {round((end - start)*1000, 2)} seconds!")
    print(f"Time elapsed: {(end - start)*1000:1.6f} Milliseconds!")

# π = 3.14159265
# Time elapsed: 112.76 seconds!

File: C/arctan_pi.c
/*
* calcul de pi formule le
gcc -O2 -Wall -o arctanpi arctan_pi.c -lm
*/

#include <stdio.h>
#include <stdlib.h>
#include <math.h>


long double calculate_pi()
{
double un ;
double trois ;
long double six ;
long double pi ;


un = 1.0 ;
trois = 3.0 ;
six = 6.0 ;
pi = 0.0 ;

  pi = six * atan( un / sqrt(trois)) ;
  return pi ;

}

int main(int argc, char **argv)
{

  long double resultat ;

 resultat = 0.0;
	resultat = calculate_pi();
  printf("%.25Lf", resultat);
return 0;
}
